### PR TITLE
Make `relativeClassName` return class name relative to models namespace

### PR DESCRIPTION
**Pull request by promatik**
This is a fix to #490.

This makes `relativeClassName` return the classname relative to models namespace.

The tests are ok, as far as I know this is only used to generate relationships of new entries, and that is working too.

This function was a little bit nonsense, because it first checked if config was set, if so, return, otherwise it would return a string where it would try to replace/remove the config, but there the config was never set anyway. I believe this was introduced to overcome the issue that was fixed on #477.

---

Before this PR, with `'models_namespace' => 'Models'` config, `Trace` would generate the following `.blueprint`;

```yaml
models:
    Models\Address: { street: 'string nullable', country: string, icon_id: 'integer nullable', monster_id: integer }
    Models\Icon: { name: string, icon: string }
```

With this, the paths are relative to `models_namepsace` config;

``` yaml
models:
    Address: { street: 'string nullable', country: string, icon_id: 'integer nullable', monster_id: integer }
    Icon: { name: string, icon: string }
```

This will avoid the "double" `Model\` on relationships namespaces.

### CODE REVIEW

**Comment by tabacitu** [comment]
**IMPORTANT! This works, but @jasonmccreary  might consider it a breaking change!** 

---

All tests below are with `'models_namespace' => 'Models'` inside the config. That's the only case where things are different.

On both `v1.24.1` and `master` I've run `php artisan blueprint:trace` and I got:
```php
models:
    Models\Address: { street: 'string nullable', country: string, icon_id: 'integer nullable', monster_id: integer }
    Models\Album: { artist_id: 'biginteger unsigned', name: string, cover: string }
    Models\Artist: { name: string, image: 'string nullable' }
```

After this PR, I've run `php artisan blueprint:trace` and I got:
```php
models:
    Address: { street: 'string nullable', country: string, icon_id: 'integer nullable', monster_id: integer }
    Album: { artist_id: 'biginteger unsigned', name: string, cover: string }
    Artist: { name: string, image: 'string nullable' }
```

Which I know, is the whole point of the PR. But still... **the output of the `trace` command is now different**! 
It _needs_ to be different on `master`, because commit https://github.com/laravel-shift/blueprint/commit/e446a418c28539f5cd0b6404692dd20930523780 only solves half the problem. But still.. it's different.

-----

The scenario where this might be considered a breaking change is when:
- a dev runs the `trace` command
- then switches from `v1.24.1` to `dev-master`
- then does NOT run `trace` again, but tries to do things with the file traced using `v1.24.1`

**I think that scenario is unrealistic, so I wouldn't consider this a breaking change.** I'd rather those very few people have a problem, rather than _everybody using `master`_ which I imagine will be more and more people. So... 

---

I would merge this - **I've just tested it and it works as expected** (noting the output difference above). 

Good find @promatik ! 👏👏👏

**Comment by jasonmccreary** [comment]
@promatik, I'd like to see some kind of test for this. Something to help confirm this won't regress again.

@tabacitu, appreciate the double check, but your _edited_ comment chain is very hard to follow. Consider using ~~strikethrough~~ or leaving individual comments to naturally relay the progression.

**Comment by jasonmccreary** [comment]
@promatik, can you add a test case for this?

**Comment by promatik** [comment]
Hi @jasonmccreary!
Sorry for the delay, I added a test for `relativeClassName()`, the function is private, so I had to use reflection to test it.
Let me know if it's ok.

**Comment by jasonmccreary** [comment]
@promatik, I was looking for a way to test this through a draft file. I don't test private functions directly. However, I already merged it, so I'll see how I might be able to backfill one. 👍
